process/block/metablock.go

```diff
@@ -1185,6 +1185,12 @@ func (mp *metaProcessor) updateState(lastMetaBlock data.HeaderHandler) {
 		return
 	}
 
+	if lastMetaBlock.IsStartOfEpochBlock() {
```

Comment:
Also here this snapshot could be executed on the same root hash more than once if a roll back occurs.

Reply:
The second time the snapshot for the same root hash occurs, no snapshot will be done.